cmd/provider/main.go

```diff
@@ -122,13 +168,11 @@ func main() {
 			Features:                &feature.Flags{},
 			MetricOptions:           &mo,
 		},
-		Provider: provider,
-		// use the following WorkspaceStoreOption to enable the shared gRPC mode
-		// terraform.WithProviderRunner(terraform.NewSharedProvider(log, os.Getenv("TERRAFORM_NATIVE_PROVIDER_PATH"), terraform.WithNativeProviderArgs("-debuggable")))
-		WorkspaceStore:        terraform.NewWorkspaceStore(log),
-		SetupFn:               clients.TerraformSetupBuilder(provider.TerraformProvider),
+		Provider:              provider,
+		SetupFn:               clients.TerraformSetupBuilder(setupCfg),
```

Comment:
When I add `WorkspaceStore: terraform.NewWorkspaceStore(log)`:

```
crossplane-p… │ 2024-07-23T20:31:30-04:00	INFO	Starting workers	{"controller": "managed/network.equinix.jet.crossplane.io/v1alpha1, kind=file", "worker count": 10}
crossplane-p… │ 2024-07-23T20:31:30-04:00	DEBUG	provider-jet-equinix	Cannot connect to provider	{"controller": "managed/metal.equinix.jet.crossplane.io/v1alpha1, kind=vrf", "request": {"name":"vrf-via-crossplane-xrd"}, "uid": "5c6b31e0-2b52-48cd-89ac-33af47e748e2", "version": "232369", "external-name": "", "error": "cannot get a terraform workspace for resource: cannot check if a Terraform dependency update is required: cannot get provider configuration", "errorVerbose": "cannot get provider configuration\ngithub.com/crossplane/upjet/pkg/terraform.(*FileProducer).needProviderUpgrade\n\tgithub.com/crossplane/upjet@v1.4.0/pkg/terraform/files.go:312\ngithub.com/crossplane/upjet/pkg/terraform.(*WorkspaceStore).Workspace\n\tgithub.com/crossplane/upjet@v1.4.0/pkg/terraform/store.go:253\ngithub.com/crossplane/upjet/pkg/controller.(*Connector).Connect\n\tgithub.com/crossplane/upjet@v1.4.0/pkg/controller/external.go:116\ngithub.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*NopDisconnecter).Connect\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/reconciler/managed/reconciler.go:246\ngithub.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*Reconciler).Reconcile\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/reconciler/managed/reconciler.go:892\ngithub.com/crossplane/crossplane-runtime/pkg/ratelimiter.(*Reconciler).Reconcile\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/ratelimiter/reconciler.go:54\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:119\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:316\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:227\nruntime.goexit\n\truntime/asm_amd64.s:1695\ncannot check if a Terraform dependency update is required\ngithub.com/crossplane/upjet/pkg/terraform.(*WorkspaceStore).Workspace\n\tgithub.com/crossplane/upjet@v1.4.0/pkg/terraform/store.go:255\ngithub.com/crossplane/upjet/pkg/controller.(*Connector).Connect\n\tgithub.com/crossplane/upjet@v1.4.0/pkg/controller/external.go:116\ngithub.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*NopDisconnecter).Connect\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/reconciler/managed/reconciler.go:246\ngithub.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*Reconciler).Reconcile\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/reconciler/managed/reconciler.go:892\ngithub.com/crossplane/crossplane-runtime/pkg/ratelimiter.(*Reconciler).Reconcile\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/ratelimiter/reconciler.go:54\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:119\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:316\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:227\nruntime.goexit\n\truntime/asm_amd64.s:1695\ncannot get a terraform workspace for resource\ngithub.com/crossplane/upjet/pkg/controller.(*Connector).Connect\n\tgithub.com/crossplane/upjet@v1.4.0/pkg/controller/external.go:118\ngithub.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*NopDisconnecter).Connect\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/reconciler/managed/reconciler.go:246\ngithub.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*Reconciler).Reconcile\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/reconciler/managed/reconciler.go:892\ngithub.com/crossplane/crossplane-runtime/pkg/ratelimiter.(*Reconciler).Reconcile\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/ratelimiter/reconciler.go:54\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:119\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:316\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:227\nruntime.goexit\n\truntime/asm_amd64.s:1695"}
crossplane-p… │ 2024-07-23T20:31:30-04:00	DEBUG	provider-jet-equinix	Cannot connect to provider	{"controller": "managed/metal.equinix.jet.crossplane.io/v1alpha1, kind=gateway", "request": {"name":"vrf-via-crossplane-xrd"}, "uid": "e2104592-ab1e-4ae0-aea3-2a78ebbcafd8", "version": "232371", "external-name": "", "error": "cannot get a terraform workspace for resource: cannot check if a Terraform dependency update is required: cannot get provider configuration", "errorVerbose": "cannot get provider configuration\ngithub.com/crossplane/upjet/pkg/terraform.(*FileProducer).needProviderUpgrade\n\tgithub.com/crossplane/upjet@v1.4.0/pkg/terraform/files.go:312\ngithub.com/crossplane/upjet/pkg/terraform.(*WorkspaceStore).Workspace\n\tgithub.com/crossplane/upjet@v1.4.0/pkg/terraform/store.go:253\ngithub.com/crossplane/upjet/pkg/controller.(*Connector).Connect\n\tgithub.com/crossplane/upjet@v1.4.0/pkg/controller/external.go:116\ngithub.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*NopDisconnecter).Connect\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/reconciler/managed/reconciler.go:246\ngithub.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*Reconciler).Reconcile\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/reconciler/managed/reconciler.go:892\ngithub.com/crossplane/crossplane-runtime/pkg/ratelimiter.(*Reconciler).Reconcile\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/ratelimiter/reconciler.go:54\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:119\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:316\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:227\nruntime.goexit\n\truntime/asm_amd64.s:1695\ncannot check if a Terraform dependency update is required\ngithub.com/crossplane/upjet/pkg/terraform.(*WorkspaceStore).Workspace\n\tgithub.com/crossplane/upjet@v1.4.0/pkg/terraform/store.go:255\ngithub.com/crossplane/upjet/pkg/controller.(*Connector).Connect\n\tgithub.com/crossplane/upjet@v1.4.0/pkg/controller/external.go:116\ngithub.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*NopDisconnecter).Connect\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/reconciler/managed/reconciler.go:246\ngithub.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*Reconciler).Reconcile\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/reconciler/managed/reconciler.go:892\ngithub.com/crossplane/crossplane-runtime/pkg/ratelimiter.(*Reconciler).Reconcile\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/ratelimiter/reconciler.go:54\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:119\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:316\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:227\nruntime.goexit\n\truntime/asm_amd64.s:1695\ncannot get a terraform workspace for resource\ngithub.com/crossplane/upjet/pkg/controller.(*Connector).Connect\n\tgithub.com/crossplane/upjet@v1.4.0/pkg/controller/external.go:118\ngithub.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*NopDisconnecter).Connect\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/reconciler/managed/reconciler.go:246\ngithub.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*Reconciler).Reconcile\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/reconciler/managed/reconciler.go:892\ngithub.com/crossplane/crossplane-runtime/pkg/ratelimiter.(*Reconciler).Reconcile\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/ratelimiter/reconciler.go:54\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:119\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:316\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:227\nruntime.goexit\n\truntime/asm_amd64.s:1695"}
crossplane-p… │ 2024-07-23T20:31:30-04:00	DEBUG	provider-jet-equinix	Cannot connect to provider	{"controller": "managed/metal.equinix.jet.crossplane.io/v1alpha1, kind=reservedipblock", "request": {"name":"vrf-via-crossplane-xrd"}, "uid": "ebe14ed6-83f3-43c7-821c-3c4268e16cfc", "version": "232373", "external-name": "", "error": "cannot get a terraform workspace for resource: cannot check if a Terraform dependency update is required: cannot get provider configuration", "errorVerbose": "cannot get provider configuration\ngithub.com/crossplane/upjet/pkg/terraform.(*FileProducer).needProviderUpgrade\n\tgithub.com/crossplane/upjet@v1.4.0/pkg/terraform/files.go:312\ngithub.com/crossplane/upjet/pkg/terraform.(*WorkspaceStore).Workspace\n\tgithub.com/crossplane/upjet@v1.4.0/pkg/terraform/store.go:253\ngithub.com/crossplane/upjet/pkg/controller.(*Connector).Connect\n\tgithub.com/crossplane/upjet@v1.4.0/pkg/controller/external.go:116\ngithub.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*NopDisconnecter).Connect\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/reconciler/managed/reconciler.go:246\ngithub.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*Reconciler).Reconcile\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/reconciler/managed/reconciler.go:892\ngithub.com/crossplane/crossplane-runtime/pkg/ratelimiter.(*Reconciler).Reconcile\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/ratelimiter/reconciler.go:54\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:119\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:316\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:227\nruntime.goexit\n\truntime/asm_amd64.s:1695\ncannot check if a Terraform dependency update is required\ngithub.com/crossplane/upjet/pkg/terraform.(*WorkspaceStore).Workspace\n\tgithub.com/crossplane/upjet@v1.4.0/pkg/terraform/store.go:255\ngithub.com/crossplane/upjet/pkg/controller.(*Connector).Connect\n\tgithub.com/crossplane/upjet@v1.4.0/pkg/controller/external.go:116\ngithub.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*NopDisconnecter).Connect\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/reconciler/managed/reconciler.go:246\ngithub.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*Reconciler).Reconcile\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/reconciler/managed/reconciler.go:892\ngithub.com/crossplane/crossplane-runtime/pkg/ratelimiter.(*Reconciler).Reconcile\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/ratelimiter/reconciler.go:54\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:119\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:316\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:227\nruntime.goexit\n\truntime/asm_amd64.s:1695\ncannot get a terraform workspace for resource\ngithub.com/crossplane/upjet/pkg/controller.(*Connector).Connect\n\tgithub.com/crossplane/upjet@v1.4.0/pkg/controller/external.go:118\ngithub.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*NopDisconnecter).Connect\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/reconciler/managed/reconciler.go:246\ngithub.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*Reconciler).Reconcile\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/reconciler/managed/reconciler.go:892\ngithub.com/crossplane/crossplane-runtime/pkg/ratelimiter.(*Reconciler).Reconcile\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/ratelimiter/reconciler.go:54\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:119\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:316\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:227\nruntime.goexit\n\truntime/asm_amd64.s:1695"}
crossplane-p… │ 2024-07-23T20:31:30-04:00	DEBUG	provider-jet-equinix	Cannot connect to provider	{"controller": "managed/metal.equinix.jet.crossplane.io/v1alpha1, kind=vlan", "request": {"name":"vrf-via-crossplane-xrd"}, "uid": "f6b1855c-8805-4755-997c-88d60e7554b3", "version": "232372", "external-name": "", "error": "cannot get a terraform workspace for resource: cannot check if a Terraform dependency update is required: cannot get provider configuration", "errorVerbose": "cannot get provider configuration\ngithub.com/crossplane/upjet/pkg/terraform.(*FileProducer).needProviderUpgrade\n\tgithub.com/crossplane/upjet@v1.4.0/pkg/terraform/files.go:312\ngithub.com/crossplane/upjet/pkg/terraform.(*WorkspaceStore).Workspace\n\tgithub.com/crossplane/upjet@v1.4.0/pkg/terraform/store.go:253\ngithub.com/crossplane/upjet/pkg/controller.(*Connector).Connect\n\tgithub.com/crossplane/upjet@v1.4.0/pkg/controller/external.go:116\ngithub.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*NopDisconnecter).Connect\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/reconciler/managed/reconciler.go:246\ngithub.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*Reconciler).Reconcile\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/reconciler/managed/reconciler.go:892\ngithub.com/crossplane/crossplane-runtime/pkg/ratelimiter.(*Reconciler).Reconcile\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/ratelimiter/reconciler.go:54\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:119\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:316\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:227\nruntime.goexit\n\truntime/asm_amd64.s:1695\ncannot check if a Terraform dependency update is required\ngithub.com/crossplane/upjet/pkg/terraform.(*WorkspaceStore).Workspace\n\tgithub.com/crossplane/upjet@v1.4.0/pkg/terraform/store.go:255\ngithub.com/crossplane/upjet/pkg/controller.(*Connector).Connect\n\tgithub.com/crossplane/upjet@v1.4.0/pkg/controller/external.go:116\ngithub.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*NopDisconnecter).Connect\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/reconciler/managed/reconciler.go:246\ngithub.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*Reconciler).Reconcile\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/reconciler/managed/reconciler.go:892\ngithub.com/crossplane/crossplane-runtime/pkg/ratelimiter.(*Reconciler).Reconcile\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/ratelimiter/reconciler.go:54\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:119\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:316\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:227\nruntime.goexit\n\truntime/asm_amd64.s:1695\ncannot get a terraform workspace for resource\ngithub.com/crossplane/upjet/pkg/controller.(*Connector).Connect\n\tgithub.com/crossplane/upjet@v1.4.0/pkg/controller/external.go:118\ngithub.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*NopDisconnecter).Connect\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/reconciler/managed/reconciler.go:246\ngithub.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*Reconciler).Reconcile\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/reconciler/managed/reconciler.go:892\ngithub.com/crossplane/crossplane-runtime/pkg/ratelimiter.(*Reconciler).Reconcile\n\tgithub.com/crossplane/crossplane-runtime@v1.16.0/pkg/ratelimiter/reconciler.go:54\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:119\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:316\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tsigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:227\nruntime.goexit\n\truntime/asm_amd64.s:1695"}
crossplane-p… │ 2024-07-23T20:31:30-04:00	DEBUG	events	cannot get a terraform workspace for resource: cannot check if a Terraform dependency update is required: cannot get provider configuration	{"type": "Warning", "object": {"kind":"Vrf","name":"vrf-via-crossplane-xrd","uid":"5c6b31e0-2b52-48cd-89ac-33af47e748e2","apiVersion":"metal.equinix.jet.crossplane.io/v1alpha1","resourceVersion":"232369"}, "reason": "CannotConnectToProvider"}
```

Reply:
I think adding WorkspaceStore is the wrong approach because this is not longer present in modern providers.

Reply:
The error that I get without WorkspaceStore appears to point at a missing Workspace or filesystem. I may have left remnants of WorkspaceStore configuration in the provider causing it to take that approach.

Reply:
This was corrected with 9095a6a and 78637d8